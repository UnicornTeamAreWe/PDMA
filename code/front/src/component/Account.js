import React from "react";
import {Col, Input, Menu, Row,Button} from "antd";
import {AlipayCircleOutlined, WeiboCircleOutlined, CodeSandboxOutlined, AntCloudOutlined} from '@ant-design/icons';

export class Account extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            data: [],
            select: '1'
        }
    }

    navTransfer = (e) => {
        this.setState({
            select: e.key,
        })
    }

    Airpay = () => {
        return (<span>放二维码处</span>)
    }
    Weibo = () => {
        return (
            <div style={{margin:50}}>
                <span>放二维码处</span>
                <Input placeholder={'username'}/>
                <Input placeholder={'password'}/>
                <Button>提交</Button>
            </div>
        )
    }
    Meituan = () => {
        return (
            <div>
                <Input placeholder={'验证码'}/>
                <Button>提交</Button>
            </div>
        )
    }
    CloudMusic = () => {
        return (
            <div>
                <span>放二维码处</span>
                <Input placeholder={'username'}/>
                <Input placeholder={'password'}/>
                <Button>提交</Button>
            </div>
        )
    }
    Content = (key) => {
        switch (key) {
            case '1':
                return this.Airpay()
            case '2':
                return this.Weibo()
            case '3':
                return this.Meituan()
            case '4':
                return this.CloudMusic()
        }
    }


    render() {
        return (
            <Row>
                <Col span={4}>
                    <Menu mode="inline" selectedKeys={[this.state.select]} onClick={this.navTransfer}>
                        <Menu.Item key="1" icon={<AlipayCircleOutlined/>}>支付宝</Menu.Item>
                        <Menu.Item key="2" icon={<WeiboCircleOutlined/>}>微博</Menu.Item>
                        <Menu.Item key="3" icon={<CodeSandboxOutlined/>}>美团</Menu.Item>
                        <Menu.Item key="4" icon={<AntCloudOutlined/>}>网易云音乐</Menu.Item>
                    </Menu>
                </Col>
                <Col span={20}>{this.Content(this.state.select)}</Col>
            </Row>
        )
    }
}
