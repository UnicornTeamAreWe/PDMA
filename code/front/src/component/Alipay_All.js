import React from 'react';
import Plot from 'react-plotly.js';
import axios from "axios";
import qs from "qs";
import { Typography, Divider } from 'antd';

const { Title, Paragraph, Text } = Typography;

const getAlipay = async (id,setData)=>{
    axios.post("http://localhost:8088/getAlipay", qs.stringify({ id: id }))
        .then(response =>{
            setData(response.data)
        })
        .catch(error=>console.log(error));
};

export function Alipay_All() {
    const [Alipay, setAlipay] = React.useState([]);

    React.useEffect(() => {
        getAlipay(1,(data)=>setAlipay(data)).catch()
    },[]);

    console.log(Alipay);
    let Net_income = 0;

    let Income_all = 0;
    let Spend_all = 0;
    let Calculate = () => {
        Income_all = 0;
        Spend_all = 0;
        Alipay.forEach(item =>{
            if (item.income_spending != null && item.income_spending.indexOf('收入') >= 0){
                Income_all += item.amount;
            }
            else if (item.income_spending != null && item.income_spending.indexOf('支出') >= 0){
                Spend_all += item.amount;
            }
        })
        Income_all = Income_all.toFixed(2);
        Spend_all = Spend_all.toFixed(2);
    }

    Calculate();
    Net_income = Income_all - Spend_all;
    let Income_Spend_arr = ['收入','支出'];
    let Income_Spend_amount = [Income_all,Spend_all];

    // let Min_time = Alipay[0].payment_time;
    // let Max_time = Alipay[(Alipay.length-1)].payment_time;
    // console.log(Min_time);
    // console.log(Max_time);

    let data1 = [
        {
            x: Income_Spend_arr,
            y: Income_Spend_amount,
            type: 'bar'
        }
    ];

    let data2 = [
        {
            labels: Income_Spend_arr,
            values: Income_Spend_amount,
            type: 'pie'
        }
    ];

    return (
        <div>
            <Plot
                data={data1}
                layout={ {width: 600, height: 400, title: 'Alipay Info'} }
            />
            <Plot
                data={data2}
                layout={ {width: 600, height: 400, title: 'Alipay Info'} }
            />
            <Typography>
                <Title>分析报告</Title>
                <Paragraph>
                    由图中内容可知，支付宝共收入{Income_all}元, 支出{Spend_all}元, 净收入{Net_income}元。
                </Paragraph>
            </Typography>
        </div>);
}
