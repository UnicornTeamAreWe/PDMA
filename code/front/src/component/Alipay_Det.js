import React from 'react';
import Plot from 'react-plotly.js';
import axios from "axios";
import qs from "qs";
import { Typography, Divider } from 'antd';

const { Title, Paragraph, Text } = Typography;

const getAlipay = async (id,setData)=>{
    axios.post("http://localhost:8088/getAlipay", qs.stringify({ id: id }))
        .then(response =>{
            setData(response.data)
        })
        .catch(error=>console.log(error));
};

export function Alipay_Det() {
    const [Alipay, setAlipay] = React.useState([]);

    React.useEffect(() => {
        getAlipay(1,(data)=>setAlipay(data)).catch()
    },[]);

    console.log(Alipay);
    let Income_name = [];
    let Income_amount = [];

    let Spend_name = [];
    let Spend_amount = [];

    let Calculate_Income = () =>{
        let Yu = -1;
        Alipay.forEach(item =>{
            if (item.funding_state != null && item.funding_state.indexOf('交易成功') >= 0){
                if (item.income_spending != null && item.income_spending.indexOf('收入') >= 0){
                    let position;
                    if (Income_name.length === 0){
                        if (item.name != null && item.name.indexOf('余额宝') >= 0){
                            Income_name.push('余额宝收益发放');
                            Income_amount.push(item.amount);
                            Yu = Income_name.indexOf('余额宝收益发放');

                        }
                        else{
                            Income_name.push(item.name);
                            Income_amount.push(item.amount);
                        }
                    }
                    else{
                        for (position = 0; position < Income_name.length; ++position){
                            if (item.name != null && item.name.indexOf('余额宝') >= 0){
                                if (Yu >= 0){
                                    Income_amount[Yu]+=item.amount;
                                    break;
                                }
                                else{
                                    Income_name.push('余额宝收益发放');
                                    Income_amount.push(item.amount);
                                    Yu = Income_name.indexOf('余额宝收益发放');
                                    break;
                                }
                            }
                            else if (Income_name[position] === item.name){
                                Income_amount[position]+=item.amount;
                                break;
                            }
                            else if (position === Income_name.length-1){
                                if (item.name != null && item.name.indexOf('余额宝') >= 0){
                                    Income_name.push('余额宝收益发放');
                                    Income_amount.push(item.amount);
                                    Yu = Income_name.indexOf('余额宝收益发放');
                                    break;
                                }
                                else{
                                    Income_name.push(item.name);
                                    Income_amount.push(item.amount);
                                    break;
                                }
                            }
                        }
                    }
                }

            }
        });
    }

    let Calculate_Spend = () =>{
        Alipay.forEach(item =>{
            if (item.funding_state != null && item.funding_state.indexOf('交易成功') >= 0){
                if (item.income_spending != null && item.income_spending.indexOf('支出') >= 0){
                    let position;
                    if (Spend_name.length === 0){
                        Spend_name.push(item.name);
                        Spend_amount.push(item.amount);
                    }
                    else{
                        for (position = 0; position < Spend_name.length; ++position){
                           if (Spend_name[position] === item.name){
                               Spend_amount[position]+=item.amount;
                               break;
                           }
                            else if (position === Spend_name.length-1){
                                Spend_name.push(item.name);
                                Spend_amount.push(item.amount);
                                break;
                            }
                        }
                    }
                }
            }
        });
    }


    Calculate_Income();
    Calculate_Spend();

    let length1 = Income_name.length;
    let length2 = Spend_name.length;

    let Str1 = "";
    let Income = ()=>{
        let i = 0;
        Income_name.forEach(item=>{
            Str1 += Income_name[i]+'部分收入'+Income_amount[i]+'元,';
            i++;
        })
    }
    Income();

    let Str2 = "";
    let Spend = ()=>{
        let i = 0;
        Spend_name.forEach(item=>{
            Str2 += Spend_name[i]+'部分支出'+Spend_amount[i]+'元,';
            i++;
        })
    }
    Spend();

    let Sum1 = 0;
    let Sum_Income = () =>{
        Sum1 = 0;
        Income_amount.forEach(item=>{
            Sum1 += item;
        })
    }
    Sum_Income();

    let Sum2 = 0;
    let Sum_Spend = () =>{
        Sum2 = 0;
        Spend_amount.forEach(item=>{
            Sum2 += item;
        })
    }
    Sum_Spend();

    let data1 = [
        {
            labels: Income_name,
            values: Income_amount,
            type: 'pie',
        }
    ];

    let data2 = [
        {
            labels: Spend_name,
            values: Spend_amount,
            type: 'pie',
        }
    ];

    return (
        <div>
            <Plot
                data={data1}
                layout={ {width: 600, height: 400, title: 'Alipay Income Detailed Info'} }
            />
            <Plot
                data={data2}
                layout={ {width: 600, height: 400, title: 'Alipay Spend Detailed Info'} }
            />
            <Typography>
                <Title>分析报告</Title>
                <Paragraph>
                    由图中内容可知，支付宝的收入主要由{length1}部分构成, 其中{Str1} 共收入{Sum1}元。
                </Paragraph>

                <Paragraph>
                    由图中内容可知，支付宝的支出主要由{length2}部分构成, 其中{Str2} 共支出{Sum2}元。
                </Paragraph>
            </Typography>
        </div>

    );
}
