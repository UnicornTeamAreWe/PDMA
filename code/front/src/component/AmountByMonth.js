import React from 'react';
import Plot from 'react-plotly.js';
import axios from "axios";
import qs from "qs";
import {Typography} from "antd";
const { Title, Paragraph, Text } = Typography;

const getTaobaoAnalysis = async (id,setData)=>{
    axios.post("http://localhost:8088/getTaobaoAnalysis", qs.stringify({ id: id }))
        .then(response =>{
            setData(response.data)
        })
        .catch(error=>console.log(error));
};

export function AmountByMonth() {
    const [Taobao, setTaobao] = React.useState([]);

    React.useEffect(() => {
        getTaobaoAnalysis(1,(data)=>setTaobao(data)).catch()
    },[]);

    console.log(Taobao);
    let Month_arr = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    let AmountByMonth = Array.from({length:12}).fill(0)
    let Amount = () =>{
        Taobao.forEach(item =>{
            AmountByMonth[item.month-1] += item.amount;
        });
    }
    Amount();
    let maxNum = Math.max.apply(null,AmountByMonth);
    let minNum = Math.min.apply(null,AmountByMonth);
    let Max_month = AmountByMonth.findIndex((item) => item === maxNum)+1;
    let Min_month = AmountByMonth.findIndex((item) => item === minNum)+1;
    let all = eval(AmountByMonth.join("+"));
    let average = (all/12).toFixed(2);

    let data1 = [
        {
            x: Month_arr,
            y: AmountByMonth,
            type: 'bar'
        }
    ];

    let data2 = [
        {
            x: Month_arr,
            y: AmountByMonth,
            type: 'scatter'
        }
    ];

    let data3 = [
        {
            values: AmountByMonth,
            labels: Month_arr,
            type: 'pie',
            hole: .4
        }
    ];

    return (
        <div>
            <Plot
                data={data1}
                layout={ {width: 600, height: 400, title: 'Amount By Month'} }
            />
            <Plot
                data={data2}
                layout={ {width: 600, height: 400, title: 'Amount By Month'} }
            />
            <Plot
                data={data3}
                layout={ {width: 600, height: 400, title: 'Amount By Month'} }
            />
            <Typography>
                <Title>分析报告</Title>
                <Paragraph>
                    图中展示了您在淘宝上每个月消费的金额，您可以从中获知每个月消费金额的大小。由图可知，消费金额最多的月份为{Max_month}月份，共消费{maxNum}元，消费金额最少的月份为{Min_month}月份，共消费{minNum}元, 平均每个月消费{average}元。
                </Paragraph>
            </Typography>
        </div>);
}
