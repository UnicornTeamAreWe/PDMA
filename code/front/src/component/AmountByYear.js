import React from 'react';
import Plot from 'react-plotly.js';
import axios from "axios";
import qs from "qs";

const getTaobaoAnalysis = async (id,setData)=>{
    axios.post("http://localhost:8088/getTaobaoAnalysis", qs.stringify({ id: id }))
        .then(response =>{
            setData(response.data)
        })
        .catch(error=>console.log(error));
};

export function AmountByYear() {
    const [Taobao, setTaobao] = React.useState([]);

    React.useEffect(() => {
        getTaobaoAnalysis(1,(data)=>setTaobao(data)).catch()
    },[]);

    let min = Math.min.apply(Math, Taobao.map(item=> {return item.year}));
    let Year = new Date().getFullYear();
    let Gap = Year - min;
    let AmountByYear = Array.from({length:Gap+1}).fill(0)
    let year_arr = [];
    let Amount = () =>{
        Taobao.forEach(item =>{
            AmountByYear[2020-item.year] += item.amount;
        });
        for (let i = 0; i <= Gap; ++i){
            let y = Year-i;
            year_arr.push(y.toString());
        }
    }
    Amount();

    let data1 = [
        {
            x: year_arr,
            y: AmountByYear,
            type: 'bar'
        }
    ];

    let data2 = [
        {
            x: year_arr,
            y: AmountByYear,
            type: 'scatter'
        }
    ];

    let data3 = [
        {
            values: AmountByYear,
            labels: year_arr,
            type: 'pie'
        }
    ];

    return (
        <div>
            <Plot
                data={data1}
                layout={ {width: 800, height: 600, title: 'Amount By Year'} }
            />
            <Plot
                data={data2}
                layout={ {width: 800, height: 600, title: 'Amount By Year'} }
            />
            <Plot
                data={data3}
                layout={ {width: 800, height: 600, title: 'Amount By Year'} }
            />
        </div>
        );
}
