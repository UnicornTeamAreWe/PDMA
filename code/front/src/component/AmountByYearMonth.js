import React from 'react';
import Plot from 'react-plotly.js';
import axios from "axios";
import qs from "qs";

const getTaobaoAnalysis = async (id,setData)=>{
    axios.post("http://localhost:8088/getTaobaoAnalysis", qs.stringify({ id: id }))
        .then(response =>{
            setData(response.data)
        })
        .catch(error=>console.log(error));
};

export function AmountByYearMonth() {
    const [Taobao, setTaobao] = React.useState([]);

    React.useEffect(() => {
        getTaobaoAnalysis(1,(data)=>setTaobao(data)).catch()
    },[]);

    let Month_arr = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
    let min = Math.min.apply(Math, Taobao.map(item=> {return item.year}));
    let data1 = [];
    let data2 = [];
    let Year = new Date().getFullYear();
    let Gap = Year - min;
    let AmountByYearMonth = [];

    let init = () =>{
        for (let i = 0; i <= Gap; ++i){
            AmountByYearMonth[i] = [];
            for(let j = 0; j <= 12; ++j){
                AmountByYearMonth[i][j] = 0;
            }
        }
    }
    init();

    let Amount = () =>{
        Taobao.forEach(item =>{
            AmountByYearMonth[2020-item.year][item.month-1] += item.amount;
        });
    }
    Amount();

    for (let i = 0; i <= Gap; ++i){
        let y = Year - i;
        let trace = {
                x: Month_arr,
                y: AmountByYearMonth[i],
                name: y,
                type: 'bar'
            };
        data1.push(trace);
    }

    for (let i = 0; i <= Gap; ++i){
        let y = Year - i;
        let trace = {
            x: Month_arr,
            y: AmountByYearMonth[i],
            name: y,
            type: 'scatter'
        };
        data2.push(trace);
    }

    return (
        <div>
            <Plot
            data={data1}
            layout={{
                width: 800,
                height: 600,
                title: 'Amount By Year Month',
                barmode: 'group'
            }}/>

            <Plot
                data={data2}
                layout={{
                    width: 800,
                    height: 600,
                    title: 'Amount By Year Month',
                    barmode: 'group'
                }}/>
        </div>
            );
}
