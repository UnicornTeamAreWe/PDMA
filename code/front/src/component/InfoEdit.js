import React from "react";

export class InfoEdit extends React.Component{
    constructor(props) {
        super(props);
        this.state={
            data:[]
        }
    }

    render() {
        return(<span>此处应有资料编辑</span>)
    }
}
