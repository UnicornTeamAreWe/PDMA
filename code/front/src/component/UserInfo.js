import React from "react";
import {Descriptions} from "antd";

export class UserInfo extends React.Component{
    constructor(props) {
        super(props);

        this.state={
            data:[]
        }
    }

    render() {
        return(
            <Descriptions title="User Info">
                <Descriptions.Item label="UserName">{this.props.user.username}</Descriptions.Item>
                <Descriptions.Item label="E-mail">{this.props.user.email}</Descriptions.Item>
            </Descriptions>
        )
    }
}
