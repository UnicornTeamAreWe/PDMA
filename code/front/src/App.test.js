import React from 'react';
import {cleanup, render} from '@testing-library/react';
import renderer from 'react-test-renderer'
import {UserInfo} from "./component/UserInfo";
import CheckboxWithLabel from "./component/CheckboxWithLabel";
import {fireEvent} from "@testing-library/dom";



// afterEach(cleanup)

// test('Get Title PDMA', () => {
//     const {getByText} = render(<App/>);
//     const linkElement = getByText('PDMA');
//     expect(linkElement).toBeInTheDocument();
// });

// describe('test UserInfo',()=>{
//     it('should has username and email', function () {
//         let user = {
//             username:'test',
//             email:'test@test.com'
//         }
//         const {getByText} = render(<UserInfo user={user} />)
//         const username = getByText('test');
//         const email = getByText('test@test.com')
//         expect(username.textContent).toEqual('test')
//         expect(email.textContent).toEqual('test@test.com')
//     });
// })
afterEach(cleanup);

it('CheckboxWithLabel changes the text after click', () => {
    const {queryByLabelText, getByLabelText} = render(
        <CheckboxWithLabel labelOn="On" labelOff="Off" />,
    );

    expect(queryByLabelText(/off/i)).toBeTruthy();

    fireEvent.click(getByLabelText(/off/i));

    expect(queryByLabelText(/on/i)).toBeTruthy();
});
