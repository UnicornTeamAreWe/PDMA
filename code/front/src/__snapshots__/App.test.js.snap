// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo props test 1`] = `
<div
  className="ant-descriptions"
>
  <div
    className="ant-descriptions-title"
  >
    User Info
  </div>
  <div
    className="ant-descriptions-view"
  >
    <table>
      <tbody>
        <tr
          className="ant-descriptions-row"
        >
          <td
            className="ant-descriptions-item"
            colSpan={1}
          >
            <span
              className="ant-descriptions-item-label"
            >
              UserName
            </span>
            <span
              className="ant-descriptions-item-content"
            >
              test
            </span>
          </td>
          <td
            className="ant-descriptions-item"
            colSpan={2}
          >
            <span
              className="ant-descriptions-item-label"
            >
              E-mail
            </span>
            <span
              className="ant-descriptions-item-content"
            >
              test@test.com
            </span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
